jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1050,10 +1039,16 @@ public String getLocalName()
     @Override
     public int getLocalPort()
     {
-        if (_channel == null)
-            return 0;
-        InetSocketAddress local = _channel.getLocalAddress();
-        return local == null ? 0 : local.getPort();
+        if (_channel != null)
+        {
+            int localPort = _channel.getLocalPort();
+            if (localPort > 0)
+                return localPort;
+            InetSocketAddress local = _channel.getLocalAddress();
+            if (local != null)
+                return local.getPort();
```

Comment:
We should not have to do this fall back yet again here.   Either is a local authority (so we have a port) or there is not (and we don't)

Reply:
Tried this.
Going this route breaks expectations in ForwardRequestCustomizer and DetectorConnection and ProxyEndPoint behaviors, so I reverted it.